qiwis.py

```diff
@@ -303,7 +320,10 @@ def unsubscribe(self, app: str, channel: str) -> bool:
         try:
             subscribers.remove(app)
         except KeyError:
+            logger.error("The app %s tried to unsubscribe from %s, "
+                         "which it does not subscribe to", app, channel)
             return False
+        logger.info("The app %s unsubscribed from %s", app, channel)

```

Comment:
😄 Nice observation, I think it is a matter of taste.
For me, the expression "subscribe to A" refers to a certain continued _status_, so since `subscribe()` is called, the app subscribes to the channel until it unsubscribes the channel.
However, "unsubscribing" is more like an event at a specific time point, so I used the past tense.

I think "subscribing" is also used as an event just like "unsubscribing", so it is totally fine to use the past tense for `subscribe()` for consistency!
Which one do you prefer?

Reply:
I was convinced by the first explanation!
I think you can leave it as it is.